File: src/main/java/com/alexkbit/intro/ignite/service/StatisticsService.java
package com.alexkbit.intro.ignite.service;

import com.alexkbit.intro.ignite.model.TaskStatus;
import com.alexkbit.intro.ignite.repository.TaskRepository;
import groovy.util.logging.Slf4j;
import org.apache.ignite.Ignite;
import org.apache.ignite.cluster.ClusterNode;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.EnumMap;
import java.util.HashMap;
import java.util.Map;

/**
 * Service for get statistics about tasks.
 */
@Slf4j
@Service
public class StatisticsService {

    @Autowired
    private Ignite ignite;

    @Autowired
    private TaskRepository taskRepository;

    public Map<String, Integer> getNodeStat() {
        Map<String, Integer> stat = new HashMap<>();
        for (ClusterNode node : ignite.cluster().nodes()) {
            String nodeId = node.id().toString();
            int count = taskRepository.countByNodeId(nodeId);
            stat.put(nodeId, count);
        }
        return stat;
    }

    public Map<TaskStatus, Integer> getStatusStat() {
        Map<TaskStatus, Integer> stat = new EnumMap<>(TaskStatus.class);
        for (TaskStatus status : TaskStatus.values()) {
            stat.put(status, taskRepository.countByStatus(status));
        }
        return stat;
    }
}

File: src/main/resources/static/js/service/statisticsService.js
'use strict';

App.factory('StatisticsService', ['$http', '$q', function($http, $q){

    var HOST = "/stat/";

    return {

        getNodeStat: function() {
            return $http.get(HOST + 'node')
                .then(
                    function(response){
                        return response.data;
                    },
                    function(errResponse){
                        console.error('Error while load statistics about nodes');
                        return $q.reject(errResponse);
                    }
                );
        },

        getStatusStat: function() {
            return $http.get(HOST + 'status')
                .then(
                    function(response){
                        return response.data;
                    },
                    function(errResponse){
                        console.error('Error while load statistics about status');
                        return $q.reject(errResponse);
                    }
                );
        }

    };

}]);
